### PR TITLE
[Num] Added a function in NaturalCoordinatesMapping.cpp to check the determinant of Jacobian.

### PR DESCRIPTION
**Pull request by wenqing**
The function gives more detailed message about the reason to failure.  Besides, the checking of the determinant of Jacobian  is moved to position before using the determinant.

### CODE REVIEW

File: NumLib/Fem/CoordinatesMapping/NaturalCoordinatesMapping.cpp [hunk 1]
```diff
@@ -96,6 +96,28 @@ computeMappingMatrices(
 {
 }
 
+static void checkJacobianDeterminant(const double detJ)
+{
+    if (detJ <= .0)
+    {
+        if (detJ == .0)
```
Comment (endJunction) [inline NumLib/Fem/CoordinatesMapping/NaturalCoordinatesMapping.cpp]: The cases could be simplified while reducing nestedness:
```c++
if (detJ < 0)
   OGS_FATAL(...);
if (detJ == 0)
   OGS_FATAL(...);
```

File: NumLib/Fem/CoordinatesMapping/NaturalCoordinatesMapping.cpp [hunk 2]
```diff
@@ -96,6 +96,28 @@ computeMappingMatrices(
...
+        {
+            OGS_FATAL(
+                "det J is zero. Please check whether:\n"
+                "\t element nodes may have the same coordinates,\n"
+                "\t or the coordinates of all nodes are not given in x "
+                "component for 1D problem,\n"
+                "\t or the coordinates of all nodes are not given in x-y "
```
Comment (endJunction) [inline NumLib/Fem/CoordinatesMapping/NaturalCoordinatesMapping.cpp]: by 'x component' and 'x-y component' you mean the x-axis and x-y-plane?

**Comment by endJunction** [comment]
@wenqing please have a look at PR https://github.com/wenqing/ogs/pull/14 which relates to the above comments.

File: NumLib/Fem/CoordinatesMapping/NaturalCoordinatesMapping.cpp (hunk 2 above)
Comment (wenqing) [reply NumLib/Fem/CoordinatesMapping/NaturalCoordinatesMapping.cpp]: Yes, it is (as your changes in your PR).

**Pull request merged**